shell/platform/android/flutter_main.cc

```diff
@@ -44,7 +45,8 @@ void FlutterMain::Init(JNIEnv* env,
                        jclass clazz,
                        jobject context,
                        jobjectArray jargs,
-                       jstring bundlePath) {
+                       jstring bundlePath,
+                       jstring appRootPath) {
```

Comment:
Who has write access to `appRootPath`? Could application code accidentally clobber the callback cache?

Reply:
This parameter is only visible to the internals of `FlutterMain.java` which only sets it to the result of `PathUtils.getFilesDir(applicationContext);` so it shouldn't be possible for the application to clobber the cache unless it's really trying to (e.g., by deleting the cache file directly).